presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpiller.java

```diff
@@ -148,9 +148,11 @@ private void writePages(Iterator<Page> pageIterator)
         try {
             InputStream input = new FileInputStream(targetFileName.toFile());
             memoryContext.setBytes(BUFFER_SIZE);
-            closer.register(input);
```

Comment:
You certainly want to register `input` in global closer, just in case `PagesSerdeUtil.readPages(serde, new InputStreamSliceInput(input, BUFFER_SIZE))` throws. Leverage that `InputStream#close` is idempotent.